### PR TITLE
*/*: bugfixes

### PR DESCRIPTION
**Pull request by jsmolic**

### CODE REVIEW

File: dev-embedded/stlink/stlink-1.6.0.ebuild [hunk 1]
```diff
@@ -17,13 +17,16 @@ fi
...
 	>=dev-libs/glib-2.32.0:2
 	x11-libs/gtk+:3"
-DEPEND="${RDEPEND}
-	virtual/pkgconfig"
+DEPEND="${RDEPEND}"
+BDEPEND="virtual/pkgconfig"
+
+src_prepare() {
+	cmake_src_prepare
+}
```
Comment (SoapGentoo) [inline dev-embedded/stlink/stlink-1.6.0.ebuild:29]: drop this, and set `inherit udev xdg cmake`

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

File: dev-embedded/stlink/stlink-1.6.0.ebuild (hunk 1 above)
Comment (jsmolic) [reply dev-embedded/stlink/stlink-1.6.0.ebuild]: That looks nicer, thanks

File: net-dns/https_dns_proxy/https_dns_proxy-9999.ebuild [hunk 2]
```diff
@@ -20,7 +20,8 @@ HOMEPAGE="https://github.com/aarond10/https_dns_proxy"
 LICENSE="MIT"
 SLOT="0"
 
-DEPEND="dev-libs/libev
+DEPEND="dev-cpp/gtest
+	dev-libs/libev
 	net-dns/c-ares
 	net-misc/curl[http2,ssl]"
 RDEPEND="${DEPEND}"
```
Comment (SoapGentoo) [inline net-dns/https_dns_proxy/https_dns_proxy-9999.ebuild:27]: please fix `src_install()`:

```
src_install() {
	cmake_src_install
	dobin "${BUILD_DIR}"/https_dns_proxy
}
```

**Review by SoapGentoo — CHANGES_REQUESTED** [review]
@jsmolic Drop all other commits (I've merged those), just fix `net-dns/https_dns_proxy`

File: net-dns/https_dns_proxy/https_dns_proxy-9999.ebuild (hunk 2 above)
Comment (jsmolic) [reply net-dns/https_dns_proxy/https_dns_proxy-9999.ebuild:27]: Done, thanks

File: net-dns/https_dns_proxy/https_dns_proxy-0_pre20200925.ebuild [hunk 3]
```diff
@@ -20,12 +20,13 @@ HOMEPAGE="https://github.com/aarond10/https_dns_proxy"
 LICENSE="MIT"
 SLOT="0"
 
-DEPEND="dev-libs/libev
+DEPEND="dev-cpp/gtest
```
Comment (SoapGentoo) [inline net-dns/https_dns_proxy/https_dns_proxy-0_pre20200925.ebuild:23]: could you work with upstream on making this optional?

**Review by SoapGentoo — APPROVED** [review]

File: net-dns/https_dns_proxy/https_dns_proxy-0_pre20200925.ebuild (hunk 3 above)
Comment (SoapGentoo) [reply net-dns/https_dns_proxy/https_dns_proxy-0_pre20200925.ebuild:23]: are you sure this is needed also at runtime, and not just for the test binaries?

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

File: net-dns/https_dns_proxy/https_dns_proxy-0_pre20200925.ebuild (hunk 3 above)
Comment (jsmolic) [reply net-dns/https_dns_proxy/https_dns_proxy-0_pre20200925.ebuild:23]: Yeah, this is just for test binaries, I'll go ahead and fix it now

Comment (jsmolic) [reply net-dns/https_dns_proxy/https_dns_proxy-0_pre20200925.ebuild]: Pushed :)

File: net-dns/https_dns_proxy/https_dns_proxy-0_pre20200925.ebuild [hunk 4]
```diff
@@ -20,12 +20,23 @@ HOMEPAGE="https://github.com/aarond10/https_dns_proxy"
 LICENSE="MIT"
 SLOT="0"
 
-DEPEND="dev-libs/libev
+IUSE="test"
+RESTRICT="!test? ( test )"
+
+COMMON_DEPEND="dev-libs/libev
```
Comment (SoapGentoo) [inline net-dns/https_dns_proxy/https_dns_proxy-0_pre20200925.ebuild:26]: rename `COMMON_DEPEND` to `RDEPEND` and include that in `DEPEND`

File: net-dns/https_dns_proxy/https_dns_proxy-0_pre20200925.ebuild [hunk 5]
```diff
@@ -20,12 +20,23 @@ HOMEPAGE="https://github.com/aarond10/https_dns_proxy"
...
 	net-dns/c-ares
 	net-misc/curl[http2,ssl]"
-RDEPEND="${DEPEND}"
+DEPEND="${COMMON_DEPEND}
+	test? ( dev-cpp/gtest )"
```
Comment (SoapGentoo) [inline net-dns/https_dns_proxy/https_dns_proxy-0_pre20200925.ebuild:30]: by the way, tests blow up for me, very likely due to the presence of clang-tidy. Install clang and check for yourself, I recommend disabling the clang-tidy checks and adding an otpion with upstream.

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

File: net-dns/https_dns_proxy/https_dns_proxy-0_pre20200925.ebuild (hunk 5 above)
Comment (jsmolic) [reply net-dns/https_dns_proxy/https_dns_proxy-0_pre20200925.ebuild:30]: I've disabled clang-tidy check

File: net-dns/https_dns_proxy/https_dns_proxy-0_pre20200925.ebuild (hunk 4 above)
Comment (jsmolic) [reply net-dns/https_dns_proxy/https_dns_proxy-0_pre20200925.ebuild]: Done

File: net-dns/https_dns_proxy/https_dns_proxy-9999.ebuild [hunk 6]
```diff
@@ -20,12 +20,25 @@ HOMEPAGE="https://github.com/aarond10/https_dns_proxy"
...
-DEPEND="dev-libs/libev
+IUSE="test"
+RESTRICT="!test? ( test )"
+
+RDEPEND="dev-libs/libev
 	net-dns/c-ares
 	net-misc/curl[http2,ssl]"
-RDEPEND="${DEPEND}"
+DEPEND="${RDEPEND}
+	test? ( dev-cpp/gtest )"
```
Comment (SoapGentoo) [inline net-dns/https_dns_proxy/https_dns_proxy-9999.ebuild:30]:
```
>>> Test phase: net-dns/https_dns_proxy-0_pre20200925
debug: cmake_src_test: entering function, parameters: 
 * Working in BUILD_DIR: "/var/tmp/portage/net-dns/https_dns_proxy-0_pre20200925/work/https_dns_proxy-0_pre20200925_build"
ctest -j 30 --test-load 999
Test project /var/tmp/portage/net-dns/https_dns_proxy-0_pre20200925/work/https_dns_proxy-0_pre20200925_build
No tests were found!!!
 * Tests succeeded.
>>> Completed testing net-dns/https_dns_proxy-0_pre20200925
```
:thinking:

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

File: net-dns/https_dns_proxy/https_dns_proxy-9999.ebuild (hunk 6 above)
Comment (jsmolic) [reply net-dns/https_dns_proxy/https_dns_proxy-9999.ebuild:30]: @SoapGentoo Whoops, sorry, that was a long day :/
There are actually no test implementations right now, yet cmake checks for gtest. I'm on it now, will disable redundant package checks :)

Comment (jsmolic) [reply net-dns/https_dns_proxy/https_dns_proxy-9999.ebuild]: Done

**Pull request closed without merging**